datahub/main.py

```diff
@@ -208,19 +209,24 @@ def get_dsr_data(
                 message = "One or more of the specified columns are invalid."
                 log.error(message)
                 raise HTTPException(status_code=400, detail=message)
-
-        log.info("Filtering data by column...")
-        filtered_data = []
-        for frame in filtered_index_data:
-            filtered_keys = {}
-            for key in frame.keys():
-                if dsr_headers[key.title()] in columns:
-                    filtered_keys[key] = frame[key]
-            filtered_data.append(filtered_keys)
-
-        return ORJSONResponse({"data": filtered_data})
-
-    return ORJSONResponse({"data": filtered_index_data})
+    else:
+        columns = list(dsr_headers.values())
+
+    log.info("Filtering data by column...")
+    filtered_data = []
+    for frame in filtered_index_data:
+        filtered_keys = {}
+        for key in frame.keys():
+            if dsr_headers[key.title()] in columns:
+                array = frame[key]
+                if not isinstance(array, str) and np.issubdtype(
+                    array.dtype, np.character
+                ):
+                    array = array.astype(str).tolist()
+                filtered_keys[key] = array
```

Comment:
In the spirit of un-indenting:
```suggestion
        for key, value in frame.items():
            if not dsr_headers[key.title()] in columns:
                continue
            elif not isinstance(value, str) and np.issubdtype(
                value.dtype, np.character
            ):
                filtered_keys[key] = value.astype(str).tolist()
            else:
                filtered_keys[key] = value
```

Reply:
Looks good to me